src/hmm/hmm-topology.cc

```diff
@@ -125,6 +132,8 @@ void HmmTopology::Read(std::istream &is, bool binary) {
       entries_[i].resize(thist_sz);
       for (int32 j = 0 ; j < thist_sz; j++) {
         ReadBasicType(is, binary, &(entries_[i][j].pdf_class));
+        ReadBasicType(is, binary, &(entries_[i][j].self_loop_pdf_class));
```

Comment:
remember the reading code needs to be back compatible to previously written models.